dist2src/core.py

```diff
@@ -412,7 +447,43 @@ def rebase_patches(self, from_branch, to_branch):
             self.source_git.checkout(from_branch)
             self.source_git.repo.git.reset("--hard", "HEAD^")
 
-        self.source_git.repo.git.rebase("--root", "--onto", to_branch)
         self.source_git.checkout(to_branch)
-        self.source_git.repo.git.merge("--ff-only", "-q", from_branch)
-        self.source_git.repo.git.branch("-d", from_branch)
+        commits_to_cherry_pick = [
+            c.hexsha[:8] for c in self.source_git.repo.iter_commits(from_branch)
```

Comment:
What is the reason for truncating the commit shas?

Reply:
The output is more readable when there is any problem during the rebase.

Reply:
You mean the output from the `cherry_pick` line bellow? Is there an example output you could show?

Anyways, please place a comment explaining the reason for the truncation, as it's not obvious.

Reply:
> You mean the output from the cherry_pick line bellow?

Yes, from the command bellow.

>  Is there an example output you could show?

I am interested also in the command, not only about its output. If the command fails it's printed (with the output). I am interested also in `git cherry-pick commit1 commit2 ...` so I can check it and/or try locally.

> Anyways, please place a comment explaining the reason for the truncation, as it's not obvious.

I can add a comment but I am really not sure if this is so important that it needs commenting. It's a general knowledge that you can use any unique prefix of the commit instead. (And git itself uses 8 characters commits a lot.)

Reply:
> I can add a comment but I am really not sure if this is so important that it needs commenting. It's a general knowledge that you can use any unique prefix of the commit instead. (And git itself uses 8 characters commits a lot.)

True, though it's not that obvious why this prefix is exactly 8 characters long in this code, and whether this has any relevance somewhere else. So instead of letting people second guess, or look for an answer, it could be clarified in a one-liner comment.

Reply:
Ok, I've added a comment to make it clear.